examples/shader/overriding_material_shader.rs

```diff
@@ -0,0 +1,177 @@
+    vec4 texture_pixel_color =  texture(
+        sampler2D(ColorMaterial_texture, ColorMaterial_texture_sampler),
+        v_Uv);
+    //If the color is transparent
+    if(texture_pixel_color.a == 0.0){
+        //Get the fragment's pixel position in the texture
+        vec2 pixel = v_Uv * textureSize(sampler2D(ColorMaterial_texture, ColorMaterial_texture_sampler),0);
+        //If the pixel's position fits in the equation y=x+b where b%2==0  (a straight line of pixels from every even row, creates a checkerboard pattern)
+        if(mod(int(pixel.y)-int(pixel.x),2.0)==0.0){
+            //Color the fragment slightly white
+            texture_pixel_color = vec4(0.502,0.502,0.502,1.0);
+        }
+        else{
+            //Color the fragment even whiter
+            texture_pixel_color = vec4(0.802,0.802,0.802,1.0);
+        }
+    }
+    //Mutliply the ColorMaterial's color with: 
+    //  If the alpha is zero, by our chckerboard pattern
+    //  If the alpha is non-zero, by the texture's pixel color
+    color *= texture_pixel_color;
+# endif
+    o_Target = color;
+}
+"#;
+fn setup(
+    mut commands: Commands,
+    mut render_pipelines: ResMut<Assets<PipelineDescriptor>>,
+    mut shaders: ResMut<Assets<Shader>>,
+    mut textures: ResMut<Assets<Texture>>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+) {
+    //If the sprite's original render pipeline already exists
+    if let Some(original_sprite_render_pipeline) =
+        render_pipelines.get(bevy::sprite::SPRITE_PIPELINE_HANDLE)
+    {
+        //We can clone the pipeline so we won't have to copy the code for setting it up
+        let mut pipeline_clone = original_sprite_render_pipeline.clone();
+        //In this example, we only want to override the fragment shader and so:
+        pipeline_clone.shader_stages.fragment =
+            Some(shaders.add(Shader::from_glsl(ShaderStage::Fragment, FRAGMENT_SHADER)));
+        //Adding our custom pipeline and making it untracked so it won't get removed automatically when no sprite uses it
+        render_pipelines.set_untracked(CUSTOM_SPRITE_PIPELINE_HANDLE, pipeline_clone);
+
+        //Creating our texture
+        let texture_size = 20;
+        //The vec's size is 4 times our texture's area, since every pixel is built out of 4 u8s (R,G,B,A)
+        let mut transparent_texture_vec: Vec<u8> =
+            Vec::with_capacity(texture_size * texture_size * 4);
+        //In textures, y is the opposite of bevy's, and so we need to go in reverse (the bottom pixels have the highest y)
+        for y in (0..texture_size).rev() {
+            for x in 0..texture_size {
+                //Make the pixel green, if y is bigger than x
+                if y > x {
+                    transparent_texture_vec.push(0); //r
+                    transparent_texture_vec.push(255); //g
+                    transparent_texture_vec.push(0); //b
+                    transparent_texture_vec.push(255); //a
```

Comment:
You can simplify this four function calls with 
```rust
transparent_texture_vec.extent_from_slice(&[0, 255, 0, 255]);
```

(Also same as lines 134-137)

Reply:
Sorry for the late reply, I don't think I've gotten an alert on your comment here...
I've implemented your suggestion, thank you :D